frontend/pages/recipe/_slug/ingredient-parser.vue

```diff
@@ -140,6 +140,12 @@ export default defineComponent({
       const { data } = await api.recipes.parseIngredients(parser.value, raw);
 
       if (data) {
+        // When we send the recipe ingredient text to be parsed, we lose the reference to the original unparsed ingredient.
+        // Generally this is fine, but if the unparsed ingredient had a title, we lose it; we add back the title for each ingredient here.
+        for (let i = 0; i < recipe.value.recipeIngredient.length; i++) {
+          data[i].ingredient.title = recipe.value.recipeIngredient[i].title;
+        }
+
```

Comment:
For the sake of my sanity, could you wrap this in a try/catch for a a TypeError incase the returned list is somehow not the same length? I think we can just carry on, but I'd like to have it just in case...

```js

try {
    for (let i = 0; i < recipe.value.recipeIngredient.length; i++) {
      data[i].ingredient.title = recipe.value.recipeIngredient[i].title;
    }
} catch (TypeError) {
   console.error("Index Mismatch Error") // or some better error message
}

```